Mathlib/LinearAlgebra/Dimension/Constructions.lean

```diff
@@ -56,7 +56,15 @@ theorem LinearIndependent.sumElim_of_quotient
 @[deprecated (since := "2025-02-21")]
 alias LinearIndependent.sum_elim_of_quotient := LinearIndependent.sumElim_of_quotient
 
-theorem LinearIndepOn.union_of_quotient {M' : Submodule R M}
+theorem LinearIndepOn.union_of_quotient {s t : Set ι} {f : ι → M} (hs : LinearIndepOn R f s)
+    (ht : LinearIndepOn R (mkQ (span R (f '' s)) ∘ f) t) : LinearIndepOn R f (s ∪ t) := by
+  apply hs.union ht.of_comp
+  convert (Submodule.range_ker_disjoint ht).symm
+  · simp
+  aesop
+
+-- TODO : change the statement to use `MkQ`, and the proof to use `LinearIndepOn.union_of_quotient`.
+theorem LinearIndepOn.union_id_of_quotient {M' : Submodule R M}
```

Comment:
Why have this as a TODO and not simply do it now? Does it break much? Also, are you sure `mkQ` is a good idea? Is it in simp normal form (I guess probably it is since it's unapplied).

Reply:
[this topic](https://leanprover.zulipchat.com/#narrow/channel/287929-mathlib4/topic/Submodule.2EmkQ.20vs.20Submodule.2EQuotient.2Emk) was what I had in mind